packages/flutter_tools/test/commands.shard/permeable/create_test.dart

```diff
@@ -1702,13 +1702,20 @@ void main() {
   });
 
   testUsingContext(
-    'invokes pub offline when requested',
+    'invokes pub in online and offline modes',
```

Comment:
If this is only testing that the --offline flag is or isn't plumbed through correctly, can it use a fake filesystem and process runner instead? We can land this to get the tree green etc. but if this is just testing flag plumbing there's no reason to invoke the real `pub`, I think?

Reply:
that makes sense to me